[docs] Fix coredb contribute instructions for centos (#16074)
yugabyte/yugabyte-db: docs/content/preview/contribute/core-database/build-from-src-centos.md

However, those cases may require additional, undocumented steps.

[linuxbrew]: https://github.com/linuxbrew/brew

### Ninja (optional)

It is recommended to use [Ninja][ninja] for faster build.

```sh
sudo yum install -y ninja-build
```

[ninja]: https://ninja-build.org

### Ccache (optional)

It is recommended to use [Ccache][ccache] for faster build.

```sh
sudo yum install -y ccache
# You may also want to add the following line to your .bashrc or equivalent.
export YB_CCACHE_DIR="$HOME/.cache/yb_ccache"
```

[ccache]: https://ccache.dev

### GCC (optional)

To be able to compile with GCC, install the following packages, substituting the version number to the GCC version you plan on using.

```sh
sudo yum install -y devtoolset-11 devtoolset-11-libatomic-devel
```

### Java

{{% readfile "includes/java.md" %}}

Both requirements can be satisfied by the package manager.

```sh
sudo yum install -y java-1.8.0-openjdk rh-maven35
# You may also want to add the following line to your .bashrc or equivalent.
source /opt/rh/rh-maven35/enable
```

## Build the code

{{% readfile "includes/build-the-code.md" %}}

### Build release package (optional)

> Additional packages are needed in order to build yugabyted-ui:

Reviewer: ```suggestion
Install the following additional packages to build yugabyted-ui:
```

```sh
sudo yum install -y npm golang
```

> ulimits may need to be modified.
> For example, build may fail with "too many open files".

Reviewer: ```suggestion
You may need to modify ulimits.
For example, the build may fail with "too many open files".
```

```sh
sudo yum update -y
sudo yum groupinstall -y 'Development Tools'
sudo yum install -y centos-release-scl epel-release git libatomic libicu rsync
```

### Python 3

Python 3.7 or higher is required.
The following example installs Python 3.8.

```sh
sudo yum -y install rh-python38
# You may also want to add the following line to your .bashrc or equivalent.
source /opt/rh/rh-python38/enable
```

### CMake 3

[CMake][cmake] 3.17.3 or higher is required.
The package manager has that, but we still need to link the name `cmake` to `cmake3`.
Do similarly for `ctest`.

```sh
sudo yum install -y cmake3
sudo ln -s /usr/bin/cmake3 /usr/local/bin/cmake
sudo ln -s /usr/bin/ctest3 /usr/local/bin/ctest
```

[cmake]: https://cmake.org

### /opt/yb-build

> By default, when running build, thirdparty libraries are not built, and pre-built libraries are downloaded.

Reviewer: ```suggestion
By default, when running build, third-party libraries are not built, and pre-built libraries are downloaded.
```

We also use [Linuxbrew][linuxbrew] to provide some of the third-party dependencies on CentOS.
The build scripts automatically install these in directories under `/opt/yb-build`.
In order for the build script to write under those directories, it needs proper permissions.
One way to do that is as follows:

```sh
sudo mkdir /opt/yb-build
sudo chown "$(whoami)" /opt/yb-build
```

Alternatively, the build options `--no-download-thirdparty` and/or `--no-linuxbrew` can be specified.
However, those cases may require additional, undocumented steps.

[linuxbrew]: https://github.com/linuxbrew/brew

### Ninja (optional)

> It is recommended to use [Ninja][ninja] for faster build.

Reviewer: Most places, you can eliminate phrases like "we recommend," "it is recommended" (kill the passive voice!!!), and so on. It's pretty obvious we're recommending you do this, since it's in the instructions we're giving you. :)
```suggestion
Use [Ninja][ninja] for faster builds.
```

```sh
sudo yum install -y ninja-build
```

[ninja]: https://ninja-build.org

### Ccache (optional)

It is recommended to use [Ccache][ccache] for faster build.

```sh
sudo yum install -y ccache
```

Set `YB_CCACHE_DIR` in your `.bashrc` or equivalent:

```sh
# You may also want to add the following line to your .bashrc or equivalent.
export YB_CCACHE_DIR="$HOME/.cache/yb_ccache"
```

[ccache]: https://ccache.dev

### GCC (optional)

> To be able to compile with GCC, install the following packages, substituting the version number to the GCC version you plan on using.

Reviewer: ```suggestion
To compile with GCC, install the following packages, and adjust the version numbers to match the GCC version you plan to use.
```
